### PR TITLE
Update illumos packages URL

### LINKED ISSUE #10742: Migrate off of pkgsrc.joyent.com
A build issue popped up in https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/683 which was getting a failure when attempting to run the build-rootfs.sh script. It was failing due to an [expired cert for pkgsrc.joyent.com](https://github.com/TritonDataCenter/pkgsrc/issues/342). The build-rootfs.sh script currently references the pkgsrc.joyent.com domain:

https://github.com/dotnet/arcade/blob/5fb434d9e8b99ff4d64381ef2f0c577b314a56e2/eng/common/cross/build-rootfs.sh#L394

However, according to this [SmartOS announcement](https://smartos.topicbox.com/groups/smartos-discuss/Tc0914fe1414685b5-M516296004a2f8ea793b60932), the recommendation is to migrate off of this domain. The script should be updated accordingly.

### PR DESCRIPTION
**Pull request by am11**
Fixes #10742.

cc @mthalman, I've tested it locally to confirm that it works as expected.

### CODE REVIEW

File: eng/common/cross/build-rootfs.sh [hunk 1]
```diff
@@ -391,7 +391,7 @@ elif [[ "$__CodeName" == "illumos" ]]; then
         --with-gnu-ld --disable-nls --disable-libgomp --disable-libquadmath --disable-libssp --disable-libvtv --disable-libcilkrts --disable-libada --disable-libsanitizer \
         --disable-libquadmath-support --disable-shared --enable-tls
     make -j "$JOBS" && make install && cd ..
-    BaseUrl=https://pkgsrc.joyent.com
+    BaseUrl=https://pkgsrc.smartos.org
     if [[ "$__UseMirror" == 1 ]]; then
         BaseUrl=http://pkgsrc.smartos.skylime.net
```
Comment (am11) [inline eng/common/cross/build-rootfs.sh:396]:
```suggestion
        BaseUrl=https://pkgsrc.smartos.skylime.net
```

**Comment by am11** [comment]
cc @hoyosjs, @janvorli
* joyent server will be decommissioned at anytime, so we should update check this in soon.
* use HTTPS in mirror URL
* when using mirror URL (`--use-mirror`), I found that we are getting multiple versions of icu, updated sed script so it selects the last one.

**Review by hoyosjs — DISMISSED** [review]

File: eng/common/cross/build-rootfs.sh [hunk 2]
```diff
@@ -402,7 +402,7 @@ elif [[ "$__CodeName" == "illumos" ]]; then
     read -ra array <<<"$__IllumosPackages"
     for package in "${array[@]}"; do
         echo "Installing '$package'"
-        package="$(grep ">$package-[0-9]" All | sed -En 's/.*href="(.*)\.tgz".*/\1/p')"
+        package="$(grep ">$package-[0-9]" All | sed -En 's/.*href="(.*)\.tgz".*/\1/; $p')"
```
Comment (am11) [inline eng/common/cross/build-rootfs.sh:405]:
```suggestion
        # find last occurrence of package in listing and extract its name
        package="$(sed -En '/.*href="('"$package"'-.*).tgz".*/h;$!d;g;s//\1/p' All)"
```

**Comment by am11** [comment]
@hoyosjs, can you merge this one please?

**Review by hoyosjs — APPROVED** [review]
😁 Ah sed, always powerful yet unreadable.

**Pull request merged**